_posts/dusan_andric/2018-05-25-icgc-jupyter.md

```diff
@@ -0,0 +1,92 @@
+
+The main driving project behind our ability to deploy this new feature is JupyterHub. JupyterHub itself is a bundle of three main technologies (tornado, node-http-proxy, jupyter) which allows notebook servers to be provisioned on a per user or session basis. 
+
+The other secret sauce is that we are using the docker spawner for launching new notebook servers using a docker image that we hand rolled ourselves to include the particular customizations and features that we want. JupyterHub and the docker containers all run inside the Cancer Genome Collaboratory cloud. 
+
+We've also extended the Authorization mechanism to perform the required DACO checks for access to the ICGC dataset in addition to generalizing the whole deployment using Ansible.
+
+The Ansible deployment of this solution, from the building of docker images to the launching of VMs, has been made available as [Jukebox](https://github.com/overture-stack/Jukebox) which is part of the [Overture software stack](http://www.overture.bio/).  
+
+<figure>
+    <img src="{{site.urlimg}}dusan_andric/jupyter/jukebox.png" />
+    <figcaption>A high level overview</figcaption>
+</figure>
+
+
+## Features
+
+The most important fact to know about ICGC Jupyter Notebooks is that they run inside the Cancer Genome Collaboratory, meaning they are co-located with the ICGC dataset itself. This provides the fastest possible access to the data stored there, and we bundle the icgc-storage-client as part of the notebook server containers such that users can interact with this data. The effort has also been made such that all the advanced features of the icgc-storage-client work inside the notebook containers. For example, users will be able to use the FUSE mount capabilities of the client.
+
+<figure>
+    <img src="{{site.urlimg}}dusan_andric/jupyter/storage_jupyter.png" />
+    <figcaption>Mounting a manifest ID from the ICGC Data Portal inside Jupyter</figcaption>
+</figure>
+
+<figure>
+    <img src="{{site.urlimg}}dusan_andric/jupyter/coverage_jupyter.png" />
+    <figcaption>A simple example of plotting read coverage from a mounted file</figcaption>
+</figure>
+
+Additionally, we've also bundled the [ICGC Python API](https://icgc-python.readthedocs.io/en/develop/) for interfacing with the data annotated and indexed in the ICGC Data Portal. This API provides a nice wrapper around the REST API and a programmatic mechanism for downloading the TSV data provided by the dynamic download service. 
+
+<figure>
+    <img src="{{site.urlimg}}dusan_andric/jupyter/data_frame.png" />
+    <figcaption>Protein expression data from the ICGC Data Portal visualized as a data frame</figcaption>
+</figure>
+
+## It's a sandbox!
+
+Even though we've provided a lot of useful tools, users will still have the ability to install any missing packages they require as it is their sandbox to play with. We make no guarantees, but user experience with regard to installing python packages with pip should fairly good. 
```

Comment:
"as it is their sandbox to play with" not a huge fan of this section. Maybe:
"We did our best to provision this sandbox environment with the most useful tools, but users will still have the ability to install any missing packages they require."